akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -252,11 +253,8 @@ object ByteString {
       if (n <= 0) ByteString.empty
       else ByteString1(bytes, startIndex, Math.min(n, length))
 
-    override def slice(from: Int, until: Int): ByteString = {
-      if (from <= 0 && until >= length) this // we can do < / > since we're Compact
-      else if (until <= from) ByteString1.empty
-      else ByteString1(bytes, startIndex + from, until - from)
-    }
+    override def slice(from: Int, until: Int): ByteString =
+      drop(from).take(until - Math.max(0, from))
```

Comment:
it was discussed in the other PR: https://github.com/akka/akka/pull/21248#discussion_r76156413